spinalcordtoolbox/spinalcordtoolbox
# Adding the EPI spinal cord segmentation model to SCT

Fixes https://github.com/spinalcordtoolbox/spinalcordtoolbox/issues/4453
## Description

Opening this PR to add the 3D nnUNet model to segment spinal cord on EPI data. The trained model can be found in this release: https://github.com/sct-pipeline/fmri-segmentation/releases/tag/v0.2

## Testing the model

To test this PR, you will need to install SCT from this branch. Here are
the steps:

1. Open a terminal, then run `git clone https://github.com/spinalcordtoolbox/spinalcordtoolbox.git` to get a
copy of the source code.
2. Run `cd spinalcordtoolbox` to enter the SCT source code folder.
3. Run `git checkout rb/4453-seg-sc-epi-nnunet` to switch to this
branch.
4. Run `./install_sct` to install SCT (the installation will exist
in this same folder). Make sure that you see all green `[OK]` at the end
of the installation.
5. Run `sct_deepseg -install-task seg_sc_epi` to
download the model.

Sample image:
[spinefmri_sub-hamburgP23_task-pain_bold_0000.nii.gz](https://github.com/spinalcordtoolbox/spinalcordtoolbox/files/15097396/spinefmri_sub-hamburgP23_task-pain_bold_0000.nii.gz)

6. Download the sample image and run run `sct_deepseg -task seg_sc_epi -i <IMAGE_PATH> -c bold` to
segment an image.
